### PR TITLE
Feat/tmpz 272 implement tracking

### PR DESCRIPTION
**Pull request by josipVuko**
### Description

Added onClick handler to components for tracking implementation.

[TMPZ-272](https://nidigitalsolutions.jira.com/browse/TMPZ-272)

### Checklist

- [ ] Have you done any manual testing?
- [ ] Does it have automated tests?
- [ ] Do you need any other PRs merged before this (if so please list)?
- [ ] Do you need to update the README/Runbook
- [ ] Have you checked for [accessibility](https://nidigitalsolutions.jira.com/wiki/spaces/TNLDigital/pages/3898048523/Accessibility+Dev+Guide#Checklist)?

### Screenshots (if appropriate):

Include screenshots if needed.

[TMPZ-272]: https://nidigitalsolutions.jira.com/browse/TMPZ-272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ts-newskit/src/components/puzzles/categoryCard/index.tsx [hunk 1]
```diff
@@ -12,9 +12,21 @@ export interface CategoryCardProps {
...
 
-export const CategoryCard: FC<CategoryCardProps> = ({ type, url, Icon }) => {
+export const CategoryCard: FC<CategoryCardProps> = ({
+  type,
+  url,
+  Icon,
+  onClick
+}) => {
+  const handleClick = () => {
+    if (onClick) {
```
Comment (ivanmariic) [inline packages/ts-newskit/src/components/puzzles/categoryCard/index.tsx:25]: In my opinion `onClick && onClick()` looks nicer

File: packages/ts-newskit/src/components/puzzles/hero-banner/index.tsx [hunk 2]
```diff
@@ -165,7 +182,10 @@ export const HeroBanner: FC<HeroBannerProps> = ({
           marginBlockStart={{ xs: 'space070', sm: 'space060' }}
           marginBlockEnd={{ xs: 'space090', sm: 'space100' }}
         >
-          Already a subscriber? <a href={loginUrl}>Log In</a>
+          Already a subscriber?{' '}
```
Comment (ivanmariic) [inline packages/ts-newskit/src/components/puzzles/hero-banner/index.tsx:185]: Do you need `{' '}` ?

Comment (adamosborne-tnl) [reply packages/ts-newskit/src/components/puzzles/hero-banner/index.tsx:185]: I presume that's there. to ensure a space between the question mark and the login link

Comment (josipVuko) [reply packages/ts-newskit/src/components/puzzles/hero-banner/index.tsx:181]: That was added by "yarn fix" command to ensure spacing after formatting code to new lines.

File: packages/ts-newskit/src/components/puzzles/categoryCard/index.tsx (hunk 1 above)
Comment (josipVuko) [reply packages/ts-newskit/src/components/puzzles/categoryCard/index.tsx]: Adjusted it on all places.

**Review by ivanmariic — APPROVED** [review]

**Review by adamosborne-tnl — APPROVED** [review]

**Pull request merged**